src/core/internal/backtrace/dwarf.d

```diff
@@ -169,65 +169,90 @@ int traceHandlerOpApplyImpl(const(void*)[] callstack, scope int delegate(ref siz
 
     auto image = Image.openSelf();
 
-    int processCallstack(const(ubyte)[] debugLineSectionData)
+    // find address -> file, line mapping using dwarf debug_line
+    Array!Location locations;
+    locations.length = callstack.length;
+    foreach (size_t i; 0 .. callstack.length)
     {
-        // find address -> file, line mapping using dwarf debug_line
-        Array!Location locations;
-        locations.length = callstack.length;
-        foreach (size_t i; 0 .. callstack.length)
-        {
-            locations[i].address = callstack[i];
-            locations[i].procedure = getMangledSymbolName(frameList[i][0 .. strlen(frameList[i])]);
-        }
-
-        if (debugLineSectionData)
-            resolveAddresses(debugLineSectionData, locations[], image.baseAddress);
+        locations[i].address = callstack[i];
+        locations[i].procedure = getMangledSymbolName(frameList[i][0 .. strlen(frameList[i])]);
+    }
 
-        char[1536] buffer = void;
-        size_t bufferLength;
-        scope sink = (scope const char[] data) {
-                // We cannot write anymore
-                if (bufferLength > buffer.length)
-                    return;
+    return image.isValid
+        ? image.processDebugLineSectionData(
+            (line) => locations[].processCallstack(line, image.baseAddress, dg))
+        : locations[].processCallstack(null, image.baseAddress, dg);
+}
```

Comment:
Final cosmetic nit:

```suggestion
    return image.isValid
      ? image.processDebugLineSectionData(line =>
          locations.processCallstack(line, image.baseAddress, dg)
        )
      : locations.processCallstack(null, image.baseAddress, dg);
}
```

It's [prettier](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAzArlMMCW0AE8AzjABQCU+wAOlAE5wzp1T44C2AhgOZwB0OIgDVOAGxwATfAH42XXnwAOdCGDhEiAETgAjdNwAyOKHADKcbHiibOMTqVLiTlALwA+fKNW2rRJSrUNAGExURJOMABrR2M4ABo5Hn4dTiI4AEEJCQYNBIluckokT29caD9lVXUiENEwuyjSKHQ6hI4kvhS0zOzqvIKAbloAXxA4kAhFMqgiZFBOOhUAdwAFBYRZlE4ANwhJMZAdOgjIxlNFCONuZBg6dHiQOHYdOCzXg04obnQkgDEIOhcGC4L7IECcdAwCAHAAWMHYogA6jCcMQLmpzDNUThtqiAJ5gsAaA7GNJ0GArY7cLjIVBiNLjABWRAAHgAhY5RM6cdhwIwmWn0h7MlmmK6iOAARXQEHggrCDwudDJYJSL1EB2UxhgiMkMBhyAAHAAGcaVNKI46KMHKdRwOjbOAHACOMvglMmm3BRAAtCZXq8DgxXTgGJSeDSkHSFeM0uwcDc7g8iOKpW6nUhbvdxnYdLqJPrkAAmHPHHBObhBCDsSOPIgAVgO6DSABVODpNtGGSBtvcAJJQCQIGCmMB0HBTdKD0wwPES+VpYbDIA) this way :P

Reply:
That's very arbitrary. We have much more horizontal than vertical space on modern monitors, so needless vertical space utilization will not satisfy everyone.

Reply:
let's ignore this comment for now, I'm interested in style discussions ;)